baseplate/lib/message_queue.py

```diff
@@ -121,6 +159,195 @@ def close(self) -> None:
         self.queue.close()
 
 
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue.
+
+    Uses a simple Python Queue to store data.
+
+    """
+
+    def __init__(self, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        try:
+            message = self.queue.get(timeout=timeout)
+            self.queue.task_done()
+            return message
+        except q.Empty:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        try:
+            self.queue.put(message, timeout=timeout)
+        except q.Full:
+            raise TimedOutError
+
+
+class RemoteMessageQueue(MessageQueue):
+    """A write-only message queue that uses a remote Thrift server.
+
+    This implementation is a temporary compromise and should only be used
+    under very specific circumstances if the POSIX alternative is unavailable.
+    Specifically, using Thrift here may have significant performance and/or
+    resource impacts.
+
+    """
+
+    prom_labels = [
+        "queue_name",
+        "queue_host",
+        "queue_port",
+        "queue_max_messages",
+    ]
+
+    remote_queue_put_requests_queued = Gauge(
+        f"{PROM_PREFIX}_pending_puts_to_sidecar",
```

Comment:
```suggestion
        f"{PROM_PREFIX}_length",
```

or `_size` -- i think that's clear enough

Reply:
Please put all the metric names in plain text. THis makes them easy to find via code search. I know not all the code fits this standard, but we are trying to adhere to it now.